lib/gooddata/lcm/actions/synchronize_clients.rb

```diff
@@ -87,6 +88,30 @@ def call(params)
                    "Details: #{sync_result['links']['details']}")
             end
 
+            if keep_only_previous_masters_count.positive?
+              deleted_projects = master_projects.count - (keep_only_previous_masters_count + 1)
+
+              if deleted_projects.positive?
+                removal_master_project_ids = []
+                begin
+                  master_projects.shift(deleted_projects).tap do |project_wrappers|
+                    project_wrappers.each do |project_wrapper|
+                      project = client.projects(project_wrapper[:master_project_id])
+                      GoodData.logger.info "Segment #{segment.segment_id}: Deleting old master workspace, project: '#{project.title}', PID: (#{project.pid})."
+                      project.delete
+                      removal_master_project_ids << project_wrapper[:master_project_id]
```

Comment:
Add catch exception for each project deletion.

Reply:
updated it